pynsot/commands/cmd_protocols.py

```diff
@@ -0,0 +1,464 @@
+def list(ctx, attributes, auth_string, circuit, delimited, description, device, grep, id,
+         interface, query, site_id, type):
+    """
+    List existing Protocols for a Site.
+
+    You must provide a Site ID using the -s/--site-id option.
+
+    You must provide the protocol type.
+
+    When listing Protocols, all objects are displayed by default. You
+    optionally may lookup a single Protocols by ID using the -i/--id option.
+
+    You may look up the protocols for a single Device using -D/--device which
+    can be either a hostname or ID for a Device.
+    """
+    data = ctx.params
+    data.pop('delimited')  # We don't want this going to the server.
+
+    # If we provide ID, show more fields
+    if id is not None:
+        display_fields = VERBOSE_FIELDS
+    else:
+        display_fields = DISPLAY_FIELDS
+
+    # If we aren't passing a sub-command, just call list(), otherwise let it
+    # fallback to default behavior.
+    if ctx.invoked_subcommand is None:
+        if query is not None:
+            ctx.obj.natural_keys_by_query(data, delimited)
+        else:
+            ctx.obj.list(
+                data, display_fields=display_fields,
+                verbose_fields=VERBOSE_FIELDS
+            )
+
+# Remove
+@cli.command()
+@click.option(
+    '-i',
+    '--id',
+    metavar='ID',
+    help='Unique ID of the Protocol being deleted.',
+    required=True,
+)
+@click.option(
+    '-s',
+    '--site-id',
+    metavar='SITE_ID',
+    type=int,
+    help='Unique ID of the Site this Protocol is under.',
+    callback=callbacks.process_site_id,
+    required=True,
+)
+@click.pass_context
+def remove(ctx, id, site_id):
+    """
+    Remove a Protocol.
+
+    You must provide a Site ID using the -s/--site-id option.
+
+    When removing an Protocol, you must provide the ID of the Protocol using
+    -i/--id.
+
+    You may retrieve the ID for an Protocol by parsing it from the list of
+    Protocols for a given Site:
+
+        nsot interfaces list --site-id <site_id> | grep <protocol>
+    """
+    data = ctx.params
+    ctx.obj.remove(**data)
+
+
+# Update
+@cli.command()
+@click.option(
+    '-a',
+    '--attributes',
+    metavar='ATTRS',
+    help='A key/value pair attached to this Protocol (format: key=value).',
+    multiple=True,
+    callback=callbacks.transform_attributes,
+)
+@click.option(
+    '-u',
+    '--auth_string',
+    metavar='AUTH_STRING',
+    default='',
+    help='The authentication string (such as MD5 sum).',
+)
+@click.option(
+    '-c',
+    '--circuit',
+    metavar='CIRCUIT',
+    help='The circuit that this protocol is running over.',
+)
+@click.option(
+    '-e',
+    '--description',
+    metavar='DESCRIPTION',
+    type=str,
+    help='The description for this Protocol.',
+)
+@click.option(
+    '-D',
+    '--device',
+    metavar='DEVICE',
+    type=types.NATURAL_KEY,
+    help=(
+        'Unique ID of the Device to which this Protocol is '
+        'running on.'
+    ),
+    required=True,
```

Comment:
Device should not be required on update.

We want to be able to update a protocol by ID without having to provide the device every time. For example: `nsot protocols update -i 1 -a asn=12345`